src/chain_table.rs

```diff
@@ -94,33 +95,41 @@ impl<const TABLE_WIDTH: usize, const BLOCK_LENGTH: usize>
     ///
     /// - `rng`         : random number generator
     /// - `kwi_value`   : DEM key used to encrypt the value
-    pub fn encrypt<const KEY_LENGTH: usize, DEM: Dem<KEY_LENGTH>>(
+    pub fn encrypt(
         &self,
         rng: &mut impl CryptoRngCore,
-        kwi_value: &DEM::Key,
+        kwi_value: &Aes256GcmKey,
     ) -> Result<Vec<u8>, Error> {
-        let bytes = self.try_to_bytes()?;
-        DEM::encrypt(rng, kwi_value, &bytes, None).map_err(Error::from)
+        let nonce = Nonce::new(rng);
+        let plaintext = self.serialize()?;
+        let mut result = Vec::with_capacity(
+            nonce.as_bytes().len() + plaintext.len() + Aes256Gcm::ENCRYPTION_OVERHEAD,
```

Comment:
You are counting the nonce length twice if I'm correct: once as `nonce.as_bytes().len()` and a second time in `Aes256Gcm::ENCRYPTION_OVERHEAD`.

Reply:
correct